java/group-layers/src/main/java/com/esri/arcgisruntime/sample/grouplayers/LayersAdapter.java

```diff
@@ -162,13 +174,41 @@ private void addChildLayer(Layer childLayer) {
 
       ((LinearLayout.LayoutParams) view.getLayoutParams()).setMarginStart(
           itemView.getResources().getDimensionPixelSize(R.dimen.adapter_item_child_margin_start));
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setOnCheckedChangeListener(null);
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setChecked(childLayer.isVisible());
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setOnCheckedChangeListener(
-          (buttonView, isChecked) -> mOnLayerCheckedChangedListener.layerCheckedChanged(childLayer, isChecked));
-      ((TextView) view.findViewById(R.id.layerNameTextView)).setText(childLayer.getName());
-      if (mChildLayout.findViewWithTag(childLayer) == null) {
-        mChildLayout.addView(view);
+
+      // if this is an exclusive layer it will have a non-null mRadioGroup
+      if (mRadioGroup != null) {
+        // hide the checkbox that exists
+        CheckBox checkBox = view.findViewById(R.id.layerCheckbox);
```

Comment:
maybe move this up and use the reference to avoid the repeat calls of ` ((CheckBox) view.findViewById(R.id.layerCheckbox))...` below.

I know it's not your code, but would make it much more readable below :)

Reply:
this is a good call, I don't know why I didn't make this change when I saw it!